### DISCUSSION TITLE
file uploading

### DISCUSSION THREAD

**Discussion by lelisoAgegnehu** [remix-run/remix#5325] [Q&A]
const formData = new FormData()
 formData.set('asset', file)
 formData.set('type', 'company_license')

I need to handle file storing directory  based on above type set 

// example

let folder = {
client: "client",
profile: "profile"
}

let fileSize = {
client: 3_0000_000,
profile: 2_000_000
}

 const uploadHandler = unstable_composeUploadHandlers(
        unstable_createFileUploadHandler({
            maxPartSize: 5_000_000, // how can I change size  based on form data 'type'?
            file: ({ filename, contentType }) => filename,
            directory: `public/${folder.profile}`, // how can I make dynamic based on form data 'type'?
        }),

        unstable_createMemoryUploadHandler()
    )

**Comment by cjayalmontecm** [comment]
have you able to upload image on public folder ?

**Comment by sergiodxa** [comment]
The upload handlers doesn't allow that, but you could do the upload to a tmp directory and later move it based on another input value.
